src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -48,12 +48,26 @@ BoundExpression TranslatePattern(BoundExpression input, BoundPattern pattern)
                         input = _factory.Local(temp);
                         for (int i = 0; i < pat.Subpatterns.Length; i++)
                         {
-                            var subProperty = pat.Subpatterns[i].Property;
+                            // TODO: review and test this code path.
+                            // https://github.com/dotnet/roslyn/issues/9542
+                            var subProperty = (pat.Subpatterns[i].Member as BoundPropertyPatternMember)?.MemberSymbol;
                             var subPattern = pat.Subpatterns[i].Pattern;
-                            var subExpression =
-                                subProperty.Kind == SymbolKind.Field
-                                    ? (BoundExpression)_factory.Field(input, (FieldSymbol)subProperty)
-                                    : _factory.Call(input, ((PropertySymbol)subProperty).GetMethod);
+                            BoundExpression subExpression;
+                            switch (subProperty?.Kind)
+                            {
+                                case SymbolKind.Field:
+                                    subExpression = _factory.Field(input, (FieldSymbol)subProperty);
+                                    break;
+                                case SymbolKind.Property:
+                                    subExpression = _factory.Call(input, ((PropertySymbol)subProperty).GetMethod);
```

Comment:
Consider using LocalRewriter.MakePropertyGetAccess helper instead, or, at least, use PropertySymbol.GetOwnOrInheritedGetMethod extension method to obtain the getter as that helper does.